### PR TITLE
feat(frontend): forgot password

### PR DESCRIPTION
**Pull request by sahariardev**
Here is the UI for forgot password. Solve [issue](https://github.com/docmost/docmost/issues/182) partially
See https://github.com/docmost/docmost/pull/250 for the backend 

Screenshot 2024-09-07 at 11 09 35 PM
Screenshot 2024-09-07 at 11 09 55 PM
Screenshot 2024-09-07 at 11 11 00 PM

### CODE REVIEW

File: apps/client/src/features/auth/components/forgot-password-form.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        .min(1, {message: "Email is required"})
+        .email({message: "Invalid email address"}),
+});
+
+const stepTwoSchema = z.object({
+    email: z
+        .string()
+        .min(1, {message: "Email is required"})
+        .email({message: "Invalid email address"}),
+    code: z.string().min(1, {message: 'Code is required'}),
```
Comment (aldy505) [inline apps/client/src/features/auth/components/forgot-password-form.tsx]: I made a change on the backend to rename every occurrence of `code` to `token`. See this commit https://github.com/docmost/docmost/pull/250/commits/0f15575e6ae4321c201801609822abb5fd27413b

Comment (sahariardev) [reply apps/client/src/features/auth/components/forgot-password-form.tsx]: thanks for letting me know. Will change accrodingly

Comment (sahariardev) [reply apps/client/src/features/auth/components/forgot-password-form.tsx]: Done

**Comment by Philipinho** [comment]
Thank you for your contribution.

**Pull request merged**